File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/GameRecyclerItemFactory.java
package me.xiaopan.assemblyadaptersample.itemfactory;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import me.xiaopan.assemblyadapter.AssemblyRecyclerItem;
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory;
import me.xiaopan.assemblyadaptersample.R;
import me.xiaopan.assemblyadaptersample.bean.Game;

public class GameRecyclerItemFactory extends AssemblyRecyclerItemFactory<GameRecyclerItemFactory.GameRecyclerItem> {

    private EventListener eventListener;

    public GameRecyclerItemFactory(Context context) {
        this.eventListener = new EventProcessor(context);
    }

    @Override
    public boolean isTarget(Object data) {
        return data instanceof Game;
    }

    @Override
    public GameRecyclerItem createAssemblyItem(ViewGroup parent) {
        return new GameRecyclerItem(R.layout.list_item_game, parent);
    }

    public interface EventListener{
        void onClickIcon(int position, Game user);
        void onClickName(int position, Game user);
        void onClickLike(int position, Game user);
    }

    private static class EventProcessor implements EventListener {
        private Context context;

        public EventProcessor(Context context) {
            this.context = context;
        }

        @Override
        public void onClickIcon(int position, Game game) {
            Toast.makeText(context, "瞅这游戏这臭逼样！", Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onClickName(int position, Game game) {
            Toast.makeText(context, "原来你叫"+game.name+"啊！", Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onClickLike(int position, Game game) {
            Toast.makeText(context, "我也"+game.like+"这游戏！", Toast.LENGTH_SHORT).show();
        }
    }

    public class GameRecyclerItem extends AssemblyRecyclerItem<Game> {
        private ImageView iconImageView;
        private TextView nameTextView;
        private TextView likeTextView;

        public GameRecyclerItem(int itemLayoutId, ViewGroup parent) {
            super(itemLayoutId, parent);
        }

        @Override
        protected void onFindViews() {
            iconImageView = findViewById(R.id.image_gameListItem_icon);
            nameTextView = findViewById(R.id.text_gameListItem_name);
            likeTextView = findViewById(R.id.text_gameListItem_like);
        }

        @Override
        protected void onConfigViews(Context context) {
            iconImageView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    eventListener.onClickIcon(getLayoutPosition(), getData());
                }
            });
            nameTextView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    eventListener.onClickName(getLayoutPosition(), getData());
                }
            });
            likeTextView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    eventListener.onClickLike(getLayoutPosition(), getData());
                }
            });
        }

        @Override
        protected void onSetData(int position, Game game) {
            iconImageView.setImageResource(game.iconResId);
            nameTextView.setText(game.name);
            likeTextView.setText(game.like);
        }
    }
}

File: sample/src/main/java/me/xiaopan/assemblyadaptersample/itemfactory/GameRecyclerItemFactory.kt
package me.xiaopan.assemblyadaptersample.itemfactory

import android.content.Context
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import me.xiaopan.assemblyadapter.AssemblyRecyclerItem
import me.xiaopan.assemblyadapter.AssemblyRecyclerItemFactory
import me.xiaopan.assemblyadaptersample.R
import me.xiaopan.assemblyadaptersample.bean.Game
import me.xiaopan.ssvt.bindView

class GameRecyclerItemFactory(context: Context) : AssemblyRecyclerItemFactory<GameRecyclerItemFactory.GameRecyclerItem>() {

    private val eventListener: EventListener

    init {
        this.eventListener = EventProcessor(context)
    }

    override fun isTarget(data: Any): Boolean {
        return data is Game
    }

    override fun createAssemblyItem(parent: ViewGroup): GameRecyclerItem {
        return GameRecyclerItem(R.layout.list_item_game, parent)
    }

    interface EventListener {
        fun onClickIcon(position: Int, user: Game)
        fun onClickName(position: Int, user: Game)
        fun onClickLike(position: Int, user: Game)
    }

    private class EventProcessor(private val context: Context) : EventListener {

        override fun onClickIcon(position: Int, game: Game) {
            Toast.makeText(context, "瞅这游戏这臭逼样！", Toast.LENGTH_SHORT).show()
        }

        override fun onClickName(position: Int, game: Game) {
            Toast.makeText(context, "原来你叫" + game.name + "啊！", Toast.LENGTH_SHORT).show()
        }

        override fun onClickLike(position: Int, game: Game) {
            Toast.makeText(context, "我也" + game.like + "这游戏！", Toast.LENGTH_SHORT).show()
        }
    }

    inner class GameRecyclerItem(itemLayoutId: Int, parent: ViewGroup) : AssemblyRecyclerItem<Game>(itemLayoutId, parent) {
        val iconImageView: ImageView by bindView(R.id.image_gameListItem_icon)
        val nameTextView: TextView by bindView(R.id.text_gameListItem_name)
        val likeTextView: TextView by bindView(R.id.text_gameListItem_like)

        override fun onConfigViews(context: Context) {
            iconImageView.setOnClickListener { eventListener.onClickIcon(layoutPosition, data) }
            nameTextView.setOnClickListener { eventListener.onClickName(layoutPosition, data) }
            likeTextView.setOnClickListener { eventListener.onClickLike(layoutPosition, data) }
        }

        override fun onSetData(position: Int, game: Game) {
            iconImageView.setImageResource(game.iconResId)
            nameTextView.text = game.name
            likeTextView.text = game.like
        }
    }
}
